alicloud/service_alicloud_ram.go

```diff
@@ -311,3 +312,53 @@ func (s *RamService) WaitForRamAccessKey(id, useName string, status Status, time
 		}
 	}
 }
+
+func (s *RamService) DescribeRamUserPolicyAttachment(id string) (response *ram.ListPoliciesForUserResponse, err error) {
+	request := ram.CreateListPoliciesForUserRequest()
+	parts, err := ParseResourceId(id, 3)
+	if err != nil {
+		return nil, WrapError(err)
+	}
+	request.UserName = parts[0]
+	raw, err := s.client.WithRamClient(func(ramClient *ram.Client) (interface{}, error) {
+		return ramClient.ListPoliciesForUser(request)
+	})
+	if err != nil {
+		if RamEntityNotExist(err) {
+			return nil, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+		}
+		return nil, WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw)
+	response, _ = raw.(*ram.ListPoliciesForUserResponse)
+	return
+}
+
+func (s *RamService) WaitForRamUserPolicyAttachment(id string, status Status, timeout int) error {
+	deadline := time.Now().Add(time.Duration(timeout) * time.Second)
+	for {
+		object, err := s.DescribeRamUserPolicyAttachment(id)
+		if err != nil {
+			if NotFoundError(err) {
+				if status == Deleted {
+					return nil
+				}
+			} else {
+				return WrapError(err)
+			}
+		}
+		if len(object.Policies.Policy) > 0 {
```

Comment:
I think the line 350 should be moved into the method DescribeRamUserPolicyAttachment